packages/vsce/src/utils/profileManagement.ts

```diff
@@ -69,185 +66,158 @@ export class ProfileManagement {
     return mProfileInfo;
   }
 
-  /**
-   * Makes axios GET request with path and axios config object given
-   * @param path
-   * @param config
-   * @returns
-   */
-  public static async makeRequest(path: string, config: AxiosRequestConfig) {
-    const response = await axios.get(path, config);
-    return response;
+  public static getSessionFromProfile(profile: imperative.IProfile): imperative.Session {
+    return new imperative.Session({
+      protocol: profile.protocol,
+      hostname: profile.host,
+      port: profile.port,
+      type: "basic",
+      user: profile.user,
+      password: profile.password,
+      rejectUnauthorized: 'rejectUnauthorized' in profile ? profile.rejectUnauthorized : true,
+    });
   }
 
-  public static cmciResponseXml2Json(data: string) {
-    return JSON.parse(xml2json(data, { compact: true, spaces: 4 }));
-  }
 
   /**
    * Populates the info
    * @param profile
    * @returns Array of type InfoLoaded
    */
   public static async getPlexInfo(profile: imperative.IProfileLoaded): Promise<InfoLoaded[]> {
-    const config: AxiosRequestConfig = {
-      baseURL: `${profile.profile.protocol}://${profile.profile.host}:${profile.profile.port}/CICSSystemManagement`,
-      auth: {
-        username: profile.profile.user,
-        password: profile.profile.password,
-      },
-    };
     const infoLoaded: InfoLoaded[] = [];
-
-    https.globalAgent.options.rejectUnauthorized = profile.profile.rejectUnauthorized;
+    const session = ProfileManagement.getSessionFromProfile(profile.profile);
 
     if (profile.profile.cicsPlex) {
+      const managedRegionsJson = await getResource(session, {
+        resource: "CICSManagedRegion",
+        plex: profile.profile.cicsPlex,
+        region: profile.profile.regionName,
```

Comment:
The `IResourceParms` interface asks for these attributes which will be for each getResource method call.

```suggestion
        name: "CICSManagedRegion",
        cicsPlex: profile.profile.cicsPlex,
        regionName: profile.profile.regionName,
```

Reply:
I'm hoping to avoid changes to these interfaces (if possible) since this is the V2 (maintenance) release stream 😓

Reply:
I don't think this is _changing_ any interfaces - the `IResourceParms` interface that `getResource` asks for as a paramter has `name`, `cicsPlex` and `regionName` so presumably that's the shape we'll need to pass to use `getResource` instead of Axios. 

We're also searching here for regions within a plex, where we don't necessarily have a regionName to pass (`profile.profile.regionName` could be undefined).

The V2 implementation of `getResource` requires a regionName to be provided - the rational behind removing that required argument in V3 was so we could make requests like this; to get region / plex resources.

I'm not sure what the best solution is here. To use `getResource` everywhere in this file rather than axios, we'd need to allow `getResource` in V2 to NOT require a regionName, like V3. Making a currently mandatory field optional wouldn't be considered a breaking change I assume? As those using this version can still use it with no changes.

Reply:
Resolved in e46c3f2b0ec33b8a6f6b0a19db7690d565ae77af